source/droid/gateway/gateway.d

```diff
@@ -136,7 +157,23 @@ final class Gateway
         assert(ws_ && ws_.connected);
 
         while (ws_.waitForData()) {
-            const packet = parseMessage(ws_.receiveText());
+            auto data = "";
+
+            if (decompressor) {
+                // TODO: This may be meh.
```

Comment:
I believe `ws_.receive` and checking the opcode would be the way to do this<sup>0</sup>. Alternatively, possibly doing `receiveBinary` with `strict = false`<sup>1</sup> and checking the first byte for, say, `'{'`, would also be an indicator - providing you can still use the buffer, as it seems calling `receiveText` as a fallback would allocate again. This is just what I gather from looking at the docs for a minute though, unsure of any other implications that might have.

[0] - https://vibed.org/api/vibe.http.websockets/WebSocket.receive
[1] - https://vibed.org/api/vibe.http.websockets/WebSocket.receiveBinary